capstone/src/main/java/com/google/step/clients/ScoreClient.java

```diff
@@ -0,0 +1,160 @@
+package com.google.step.clients;
+
+import com.google.step.data.RestaurantPageViews;
+import com.google.step.data.RestaurantScore;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Collections;
+
+
+/**
+ * Will take care of calculating the score for each restaurant.
+ */
+public class ScoreClient {
+
+    private final float minimum = 10.0f;
+    private final MetricsClient metricsClient = new MetricsClient();
+
+    /**
+     * Only method to be called and calculates the scores of each restaurant.
+     * @return an unmodifiable list of restaurantScore class.
+     */
+    public List<RestaurantScore> calculateScores() {
+        Map<Long, Double> scoreMap = new HashMap<>();
+        List<Long> restaurantIds = new ArrayList<>();
+
+        List<RestaurantPageViews> allPageViews = metricsClient.getAllPageViews("restaurantKey");
+
+        double systemAverage = getSystemAverage(allPageViews);
+
+        // Calculate score
+        for (RestaurantPageViews restaurant: allPageViews) {
+            double restaurantAverage = calculateRestaurantAveragePageViews(restaurant);
+
+            // Since the data is sorted we can just get the latest pageView for that restaurant.
+            int lastPageViewIndex = restaurant.getPageViews().size() - 1;
```

Comment:
Would it make more sense to call getPageViews() only once instead?

Reply:
In this instance it is being called only once but I'll change it on the other places.

Reply:
It's being called twice in this instance too (line 38 and line 39)

Reply:
Ah true. Changed.